google/cloud/internal/grpc_request_metadata_test.cc

```diff
@@ -46,6 +47,27 @@ TEST(GrpcRequestMetadata, GetRequestMetadataFromContext) {
                                                 Pair("trailer2", "value4")));
 }
 
+TEST(GrpcRequestMetadata,
+     GetRequestMetadataFromContextWhenInitialMetadataIsNotReady) {
+  auto const server_metadata =
+      RpcMetadata{{{"header1", "value1"}, {"header2", "value2"}},
+                  {{"trailer1", "value3"}, {"trailer2", "value4"}}};
+  grpc::ClientContext context;
+  testing_util::SetServerMetadata(context, server_metadata,
+                                  /*is_initial_metadata_ready*/ false);
```

Comment:
If you want to test that this case does not crash and burn, you should remove this call.

Reply:
Done